### PR TITLE
feat(ingest): add entity registry in codegen

### PR DESCRIPTION
**Pull request by hsheth2**
This PR adds information from the entity registry to the codegen system. Specifically, it adds a get_aspect_info() method to aspects that contains the entity name, docs and category annotations if available, and list of supported aspects.

Why is this useful? It sets up some interesting use cases in the future:
1. Deeper validation of MCPs - we should error if you try to add ChartInfo to a dataset.
2. Will let us autogenerate urn classes for each entity.
3. Build generic transformers instead of ones limited to a single entity type. Our current transformers only operate on datasets, but we'd like to build ones that can, for instance, add ownership to every entity that supports ownership.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by jjoyce0510 — COMMENTED** [review]
Pretty exciting PR. Can you please add explanation in the PR description for context?

**Comment by hsheth2** [comment]
@jjoyce0510 updated the PR description on why this is useful

File: metadata-ingestion/scripts/avro_codegen.py [hunk 1]
```diff
@@ -178,56 +238,62 @@ def get_aspect_type(cls) -> str:
...
-def generate(schemas_path: str, outdir: str) -> None:
-    required_schema_files = {
-        "mxe/MetadataChangeEvent.avsc",
-        "mxe/MetadataChangeProposal.avsc",
-        "usage/UsageAggregation.avsc",
-        "mxe/MetadataChangeLog.avsc",
-        "mxe/PlatformEvent.avsc",
-        "platform/event/v1/EntityChangeEvent.avsc",
-        "metadata/query/filter/Filter.avsc",  # temporarily added to test reserved keywords support
+def generate(entity_registry: str, schemas_path: str, outdir: str) -> None:
```
Comment (jjoyce0510) [inline metadata-ingestion/scripts/avro_codegen.py:259]: When folks have custom extensions to base entity registry, is the idea that they would have merged these in advanced?

**Review by jjoyce0510 — APPROVED** [review]
Looks solid to me.

File: metadata-ingestion/scripts/avro_codegen.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/scripts/avro_codegen.py:259]: Yup

If they have custom extensions to the entity registry, they're almost certainly building this from source anyways, so it's an ok assumption to make

**Pull request merged**